crates/node-types/src/action.rs

```diff
@@ -0,0 +1,23 @@
+//! # Action
+//!
+//! Actions use predicates to generate solutions.
+
+use essential_types::{PredicateAddress, Value};
+
+/// Actions represent are used as the input to solve a predicate
+/// and generate a [`Solution`](essential_types::Solution).
+pub struct Action {
+    /// The predicate that will be solved by this action.
+    pub predicate_to_solve: PredicateAddress,
+    /// The inputs to the predicate.
+    pub inputs: Vec<Value>,
+}
```

Comment:
I'm wondering if we should make `msg_sender` a field on the `Action` type?
Might be simpler than passing the sender to the predicate separately, and would logically group each action with the entity that triggered it.

Reply:
I thought about this but it means we would have to have a special field on the `Solution` as well which is really pretty specific to this one deployment of our chain. I think it would be better to just always pass it in a the first param

Reply:
Yeah, makes sense to keep it more general-purpose for flexibility.